packages/frontend-shared/src/utils/isAllowedFeature.ts

```diff
@@ -0,0 +1,104 @@
+import interval from 'human-interval'
+import { BannerIds } from '@packages/types'
+import type { LoginConnectStore } from '../store/login-connect-store'
+
+const bannerIds = {
+  isLoggedOut: BannerIds.ACI_082022_LOGIN,
+  needsOrgConnect: BannerIds.ACI_082022_CREATE_ORG,
+  needsProjectConnect: BannerIds.ACI_082022_CONNECT_PROJECT,
+  needsRecordedRun: BannerIds.ACI_082022_RECORD,
+}
+
+/**
+ *
+ * @param eventTime - timestamp of an event
+ * @param waitTime - time to compare with, such as `1 day`, `20 minutes`, etc, to be parsed by `human-interval` package
+ */
+const minTimeSinceEvent = (eventTime: number | undefined, waitTime: string) => {
+  if (!eventTime) {
+    // if there is no event time, the event has never been recorded
+    // so the "time since" is infinite, return true here
+    return true
+  }
+
+  // converting to lowercase since `interval` will error on "Day" vs "day"
+  const waitTimestamp = interval(waitTime.toLocaleLowerCase())
+
+  if (isNaN(waitTimestamp)) {
+    throw new Error(`incorrect format for waitTime provided, value must be \`n days\`, \`n minutes\` etc. Value received was ${waitTime}`)
+  }
+
+  return (Date.now() - eventTime) > interval(waitTime)
+}
+
+export const isAllowedFeature = (
+  featureName: 'specsListBanner' | 'docsCiPrompt',
+  loginConnectStore: LoginConnectStore,
```

Comment:
I'm not sure this is all that much, but it feels slightly less coupled to the idea of "a store" this way. It's convenient for these tests to pass the real store since there's code in there we want to test, but all that's needed is really and object with the correct properties. I think for practical purposes it's just a preference.